### PR TITLE
Fix MapBox background map.

### PR DESCRIPTION
**Pull request by mdastous-bentley**

### CODE REVIEW

**Comment by pmconne** [comment]
@DStradley does ImageTests run any views with mapbox imagery?

**Comment by DStradley** [comment]
> @DStradley does ImageTests run any views with mapbox imagery?

Yes, it has several views, but mapbox has not worked since ImageTests started, so the reference does not show any differences (they are all black for the map area).

**Comment by pmconne** [comment]
> Yes, it has several views, but mapbox has not worked since ImageTests started, so the reference does not show any differences (they are all black for the map area).

@DStradley right, so if we run ImageTests against this PR we should expect (good) differences.
@mdastous-bentley will you do that please?

**Review by pmconne — APPROVED** [review]

**Comment by pmconne** [comment]
Queued ImageTests. Should be 5 differences.

**Review by bsteinbk — APPROVED** [review]

**Comment by pmconne** [comment]
ImageTests produced no differences. We were expecting the views that use mapbox imagery to start displaying correctly.
@DStradley is ImageTests using an up-to-date .env file with a valid MapBox key?

File: test-apps/display-test-app/src/frontend/App.ts [hunk 1]
```diff
@@ -235,7 +235,7 @@ export class DisplayTestApp {
         ],
         /* eslint-disable @typescript-eslint/naming-convention */
         mapLayerOptions: {
-          MapBoxImagery: configuration.mapBoxKey ? { key: "access_token", value: configuration.mapBoxKey } : undefined,
+          MapboxImagery: configuration.mapBoxKey ? { key: "access_token", value: configuration.mapBoxKey } : undefined,
```
Comment (pmconne) [inline test-apps/display-test-app/src/frontend/App.ts:238]: I don't understand why this changed case. The property in `MapLayerOptions` is `MapBoxImagery`, not `MapboxImagery`.
I think you must be looking it up by MapBoxImageryMapLayerFormat.formatId, which is "MapboxImagery".
I will rename the MapLayerOptions property accordingly.

The compiler can't catch these mistakes due to the index signature on MapLayerOptions.

Comment (pmconne) [reply test-apps/display-test-app/src/frontend/App.ts:238]: Following my change the MapBox views in ImageTests produce map imagery.

**Comment by pmconne** [comment]
> @pmconne To answer your previous question, the 'MapBoxImagery' property on the MapLayerOptions interface (MapLayerOptions:46) is not read. 

Can you please explain why changing the case (MapBox vs Mapbox) fixed anything?

> As a side not I don't understand the emergency to rush the merge of this PR

There was no emergency. All of the requirements were met, so it was merged.

**Comment by mdastous-bentley** [comment]
IT works because you made the property name 'MapboxImagery' matches the case of the formatId. 

Somehow properties are included in index signature keys, look at this [sandbox](https://www.typescriptlang.org/play?#code/JYOwLgpgTgZghgYwgAgLJwA4Bk4E9oDSEuAsAFADe5yNyA1sQFwDOYUoA5gNzW0BucADYBXCCzaceZAL7lQkWIhTpseaAHkMYYAHsQzZFTK00mAEI6AHgEkAtnA7RcAfkanV+KEVxTeZzirMru44nt5SNADaMDpQ9mBurOwgHAC6biqhhMTIAD7IwiAAJhAwoBBFUrJk5Ah6rMiQrIyZalCa2vU0ALyGvCYqAEZWdg5ObhT0TMgARIhIzMwA+mA6DCAzADTIAiJis01gM8jSm-20-imBE1O4bjMMuFs7QqL3h8fVJtK+ZHX6OkEEAAdIIdBwABSHSIzIYjeyOKBPVIASi4QA).

**Comment by aruniverse** [comment]
I dont completely understand the reasoning behind changing the Map box key from `MapBox` to `Mapbox`.

I haven't spent more than 5 mins looking at the code, but i want to say this was the [real culprit](https://github.com/iTwin/itwinjs-core/blob/master/core/common/src/MapLayerSettings.ts#L444).

cc @pmconne @mdastous-bentley

**Pull request merged**